### ISSUE TITLE
Hyperopt error Windows

### ISSUE THREAD

**Issue by shaoleen00** [freqtrade/freqtrade#1232]
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Python Version: Python 3.6.3
  * CCXT version: ccxt==1.17.283
  * Branch: Master 
  * Last Commit ID: bcf47b29ed523b71190469949e0251d748d48c07
  
## Step 3: Describe the problem:

I was trying to hyperopt my strategies, and when i'm trying this error appears.
"AttributeError: Can't get attribute 'estrategia_teste' on <module 'freqtrade.strategy' from"
I'm using windows, and i can do backtest.
Command line i'm using: "python main.py hyperopt"
  
### Observed Results:

  * What happened?
  * What did you expect to happen?

### Relevant code exceptions or logs:
```
Process SpawnPoolWorker-2:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'estrategia_teste' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\d
```

**Comment by xmatthias** [comment]
the error suggests that the strategy cannot be loaded.

The problem in debugging hyperopt is currently that it requires you to change the hyperopt-file.
If you can share the changes you made, that should help in debugging the issue.

**Comment by shaoleen00** [comment]
I'm using the default file from hyperopt and the error persist.
This is the error log from prompt.

2018-09-23 10:26:57,677 - freqtrade.configuration - INFO - Verbosity set to 0
2018-09-23 10:26:57,678 - freqtrade.configuration - INFO - Dry run is enabled
2018-09-23 10:26:57,678 - freqtrade.configuration - INFO - Using DB: "sqlite://"
2018-09-23 10:26:57,678 - freqtrade.configuration - INFO - Using max_open_trades: 10 ...
2018-09-23 10:26:57,679 - freqtrade.configuration - INFO - Using data folder: user_data\data\binance ...
2018-09-23 10:26:57,680 - freqtrade.configuration - INFO - Parameter --epochs detected ...
2018-09-23 10:26:57,680 - freqtrade.configuration - INFO - Will run Hyperopt with for 10 epochs ...
2018-09-23 10:26:57,680 - freqtrade.configuration - INFO - Parameter -s/--spaces detected: all
2018-09-23 10:26:57,681 - freqtrade.strategy.resolver - WARNING - Path "/some/folder/" does not exist
2018-09-23 10:26:57,683 - freqtrade.strategy.resolver - INFO - Using resolved strategy DefaultStrategy from 'C:\Users\Nayara Costa\Desktop\Daniel\RoboCripto\freqtrade\freqtrade\user_data\strategies'
2018-09-23 10:26:57,684 - freqtrade.strategy.resolver - INFO - Override strategy 'minimal_roi' with value in config file: {'0': 0.1}.
2018-09-23 10:26:57,684 - freqtrade.strategy.resolver - INFO - Override strategy 'stoploss' with value in config file: -0.03.
2018-09-23 10:26:57,685 - freqtrade.strategy.resolver - INFO - Override strategy 'ticker_interval' with value in config file: 5m.
2018-09-23 10:26:57,685 - freqtrade.strategy.resolver - WARNING - Path "/some/folder/" does not exist
2018-09-23 10:26:57,687 - freqtrade.strategy.resolver - INFO - Using resolved strategy DefaultStrategy from 'C:\Users\Nayara Costa\Desktop\Daniel\RoboCripto\freqtrade\freqtrade\user_data\strategies'
2018-09-23 10:26:57,688 - freqtrade.strategy.resolver - INFO - Override strategy 'minimal_roi' with value in config file: {'0': 0.1}.
2018-09-23 10:26:57,689 - freqtrade.strategy.resolver - INFO - Override strategy 'stoploss' with value in config file: -0.03.
2018-09-23 10:26:57,689 - freqtrade.strategy.resolver - INFO - Override strategy 'ticker_interval' with value in config file: 5m.
2018-09-23 10:26:57,689 - freqtrade.strategy.resolver - INFO - Override process_only_new_candles 'process_only_new_candles' with value in config file: False.
2018-09-23 10:26:57,691 - freqtrade.exchange - INFO - Instance is running with dry_run enabled
2018-09-23 10:26:58,763 - freqtrade.exchange - INFO - Using Exchange "Binance"
2018-09-23 10:27:07,207 - freqtrade.optimize.hyperopt - INFO - Found 4 CPU cores. Let's make them scream!
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Process SpawnPoolWorker-3:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'DefaultStrategy' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\daniel\\robocripto\\freqtrade\\freqtrade\\strategy\\__init__.py'>
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Process SpawnPoolWorker-2:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'DefaultStrategy' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\daniel\\robocripto\\freqtrade\\freqtrade\\strategy\\__init__.py'>
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Process SpawnPoolWorker-1:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'DefaultStrategy' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\daniel\\robocripto\\freqtrade\\freqtrade\\strategy\\__init__.py'>
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Process SpawnPoolWorker-4:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'DefaultStrategy' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\daniel\\robocripto\\freqtrade\\freqtrade\\strategy\\__init__.py'>

**Comment by xmatthias** [comment]
I'll be honest, i am not sure if hyperopt is working natively in windows.
It should work when using WLS (Windows linux subsystem) - but natively in windows, i'm absolutely not sure as i never tried.

**Comment by ynzheng** [comment]
> I am not sure if hyperopt is working natively in windows (which you appear to be using - judging from the paths).
> It should work when using WLS (Windows linux subsystem) - but natively in windows, i'm absolutely not sure as i never tried.

It works on my windows 7 laptop .
(not WLS Windows linux subsystem )
(just cmder+scoop+python3.6+freqtrade develop)

the DefaultStrategy  should be in
"freqtrade\strategy"
but on log:
"Using resolved strategy DefaultStrategy from 'C:\Users\Nayara Costa\Desktop\Daniel\RoboCripto\freqtrade\freqtrade\user_data\strategies'"

what "WARNING - Path "/some/folder/" does not exist“ ？
remove the " --strategy-path /some/folder" from your command,i think ,it should be working .

last but not the least,

do run the following:

pip install -r requirements.txt
pip install -e .

**Comment by shaoleen00** [comment]
I've installed pip install -e.
pip install -r requirements.txt

I changed my command line for this:
"python main.py -c config.json --strategy-path C:\Teste hyperopt -e 200"

And the error:

(robot) (C:\server\python\conda) C:\Users\Nayara Costa\Desktop\Daniel\RoboCripto\freqtrade\freqtrade>python main.py -c config.json --strategy-path C:\Teste hyperopt -e 200
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
2018-09-25 20:12:37,854 - freqtrade.configuration - INFO - Verbosity set to 0
2018-09-25 20:12:37,855 - freqtrade.configuration - INFO - Dry run is enabled
2018-09-25 20:12:37,855 - freqtrade.configuration - INFO - Using DB: "sqlite://"
2018-09-25 20:12:37,856 - freqtrade.configuration - INFO - Using max_open_trades: 10 ...
2018-09-25 20:12:37,856 - freqtrade.configuration - INFO - Using data folder: user_data\data\binance ...
2018-09-25 20:12:37,857 - freqtrade.configuration - INFO - Parameter --epochs detected ...
2018-09-25 20:12:37,857 - freqtrade.configuration - INFO - Will run Hyperopt with for 200 epochs ...
2018-09-25 20:12:37,858 - freqtrade.configuration - INFO - Parameter -s/--spaces detected: all
2018-09-25 20:12:37,860 - freqtrade.strategy.resolver - INFO - Using resolved strategy DefaultStrategy from 'C:\Teste'
2018-09-25 20:12:37,861 - freqtrade.strategy.resolver - INFO - Override strategy 'minimal_roi' with value in config file: {'0': 0.1}.
2018-09-25 20:12:37,861 - freqtrade.strategy.resolver - INFO - Override strategy 'stoploss' with value in config file: -0.03.
2018-09-25 20:12:37,862 - freqtrade.strategy.resolver - INFO - Override strategy 'ticker_interval' with value in config file: 5m.
2018-09-25 20:12:37,863 - freqtrade.strategy.resolver - INFO - Using resolved strategy DefaultStrategy from 'C:\Teste'
2018-09-25 20:12:37,864 - freqtrade.strategy.resolver - INFO - Override strategy 'minimal_roi' with value in config file: {'0': 0.1}.
2018-09-25 20:12:37,865 - freqtrade.strategy.resolver - INFO - Override strategy 'stoploss' with value in config file: -0.03.
2018-09-25 20:12:37,865 - freqtrade.strategy.resolver - INFO - Override strategy 'ticker_interval' with value in config file: 5m.
2018-09-25 20:12:37,865 - freqtrade.strategy.resolver - INFO - Override process_only_new_candles 'process_only_new_candles' with value in config file: False.
2018-09-25 20:12:37,866 - freqtrade.exchange - INFO - Instance is running with dry_run enabled
2018-09-25 20:12:38,923 - freqtrade.exchange - INFO - Using Exchange "Binance"
2018-09-25 20:12:45,259 - freqtrade.optimize.hyperopt - INFO - Found 4 CPU cores. Let's make them scream!
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Process SpawnPoolWorker-2:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'DefaultStrategy' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\daniel\\robocripto\\freqtrade\\freqtrade\\strategy\\__init__.py'>
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Process SpawnPoolWorker-4:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'DefaultStrategy' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\daniel\\robocripto\\freqtrade\\freqtrade\\strategy\\__init__.py'>
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Process SpawnPoolWorker-3:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'DefaultStrategy' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\daniel\\robocripto\\freqtrade\\freqtrade\\strategy\\__init__.py'>
C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d
Process SpawnPoolWorker-1:
Traceback (most recent call last):
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 258, in _bootstrap
    self.run()
  File "c:\server\python\conda\Lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\server\python\conda\Lib\multiprocessing\pool.py", line 108, in worker
    task = get()
  File "C:\Users\NAYARA~1\Desktop\Daniel\ROBOCR~1\robot\lib\site-packages\sklearn\externals\joblib\pool.py", line 362, in get
    return recv()
  File "c:\server\python\conda\Lib\multiprocessing\connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
AttributeError: Can't get attribute 'DefaultStrategy' on <module 'freqtrade.strategy' from 'c:\\users\\nayara costa\\desktop\\daniel\\robocripto\\freqtrade\\freqtrade\\strategy\\__init__.py'>

**Comment by xmatthias** [comment]
can you please post the output of the following command:
```
pip freeze | grep scikit-learn
```

if you installed scikit-lear through another method, please try to find the version-number for this.

#1240 shows that 0.20 seems to cause a problem with hyperopt.

**Comment by shaoleen00** [comment]
xmatthias i tested this pull request, and function on my machine. But only just 1 processor. Manually i changed on the hyperopt code. After i changed the code uses 4 CPU.

def get_optimizer(self, cpu_count) -> Optimizer:
        return Optimizer(
            self.hyperopt_space(),
            base_estimator="ET",
            acq_optimizer="auto",
            n_initial_points=30,
            acq_optimizer_kwargs={'n_jobs': 4}
        )

**Comment by xmatthias** [comment]
what was your `-e` parameter? 

the code does a max over available cpu's vs. number of epochs ... - and the parameter `cpu_count` should contain your cpu number (4, probably).

**Comment by xmatthias** [comment]
hi @shaoleen00, is hyperopt on windows now working with #1247 merged?

if not, please post the log / output of the (unmodified) version so we can have a look.

**Comment by xmatthias** [comment]
closing as comments in #1270 suggest hyperopt works on "plain" windows.

feel free to reopen if you still encounter the problem.

**Comment by podseb** [comment]
In file hyperopt.py just replace the following line through cpus = 1 and it will work fine. It seems to be an error associated with a run in parallel. For one processor on Ubuntu 18.04 should work fine as it is in my case

line 321 #cpus = multiprocessing.cpu_count()
              cpus = 1

**Comment by hroff-1902** [comment]
> In file hyperopt.py just replace the following line through cpus = 1 and it will work fine. It seems to be an error associated with a run in parallel. For one processor on Ubuntu 18.04 should work fine as it is in my case
> 
> line 321 #cpus = multiprocessing.cpu_count()
> cpus = 1

It's not necessary to change anything in the Freqtrade code in the case you described -- you can run hyperopt with the `-j 1` command line option...

However, better report it in a separate new issue.

**Comment by podseb** [comment]
> > In file hyperopt.py just replace the following line through cpus = 1 and it will work fine. It seems to be an error associated with a run in parallel. For one processor on Ubuntu 18.04 should work fine as it is in my case
> > line 321 #cpus = multiprocessing.cpu_count()
> > cpus = 1
> 
> It's not necessary to change anything in the Freqtrade code in the case you described -- you can run hyperopt with the `-j 1` command line option...
> 
> However, better report it in a separate new issue.

Thanks for your information, I am not expert on this but by changing this parameter, I could run the hyperopt and get results. I will check it with your option!

**Comment by xmatthias** [comment]
It seems like you're using an outdated version of freqtrade - `-j` was introduced around half a year ago.

please post the output of `freqtrade --version`.

**Comment by podseb** [comment]
> It seems like you're using an outdated version of freqtrade - `-j` was introduced around half a year ago.
> 
> please post the output of `freqtrade --version` - and if the problem persists best open a new issue - attaching to year-old issues is considered bad practice, since everyone in this issue will get notifications for new comments.

Thanks for your reply, Matthias. The version I am using is:
freqtrade 0.17.3

I much apprecitae any advice on this matter you can provide me in order to solve this problem with running it in parallel. Thanks!

**Comment by hroff-1902** [comment]
There was no version 0.17.3 released in the Freqtrade codebase.
It seems to be a version forked by the Udemy course creators, and it's more than one year old and outdated.

You should better ask Udemy for support for this version or update Freqtrade to be up-to-date from the main repo.

Since your current version is from Udemy, it's better to reinstall it -- keep your custom strategy/hyperopt in a diff. place, remove your current Freqtrade from Udemy and install the latest one from scratches as written in our docs: https://www.freqtrade.io/en/latest/installation/

**Comment by podseb** [comment]
> There was no version 0.17.3 released in the Freqtrade codebase.
> It seems to be a version forked by the Udemy course creators, and it's more than one year old and outdated.
> 
> You should better ask Udemy for support for this version or update Freqtrade to be up-to-date from the main repo.
> 
> Since your current version is from Udemy, it's better to reinstall it -- keep your custom strategy/hyperopt in a diff. place, remove your current Freqtrade from Udemy and install the latest one from scratches as written in our docs: https://www.freqtrade.io/en/latest/installation/

that is exactly the case, thanks for your advice! I will install it and then replace both the strategy and optimazation folder with the current ones

**Comment by hroff-1902** [comment]
replace only custom code in those dirs -- the samples that are the part of the freqtrade installation have also changed since that time.

**Comment by podseb** [comment]
thanks for the advice again, if there is any issue, i will get back to this post!

**Comment by hroff-1902** [comment]
> thanks for the advice again, if there is any issue, i will get back to this post!

Please no. Create a new issue, This one is an old closed one.

**Comment by hroff-1902** [comment]
You can also ask questions at our Slack channel, see the docs for an invitation link.

**Comment by podseb** [comment]
just joined, thanks for it, mate. I will use this channel for more information too
